src/Carousel/widget/Carousel.ts

```diff
@@ -0,0 +1,207 @@
+import * as dojoDeclare from "dojo/_base/declare";
+import * as _WidgetBase from "mxui/widget/_WidgetBase";
+
+import { createElement } from "Carousel/lib/react";
+import { render, unmountComponentAtNode } from "Carousel/lib/react-dom";
+
+import { HeightUnits, ImageSource, OnClickEvent,
+    PageLocation, StaticImageCollection, WidthUnits } from "../Carousel.d";
+import { ImageCarousel, ImageCarouselProps } from "./components/ImageCarousel";
+
+export interface Data {
+    caption?: string;
+    description?: string;
+    guid?: string;
+}
+
+export class CarouselWrapper extends _WidgetBase {
+    // Require context will be overwritten with "NoContext" version of the widget.
+    private requiresContext: boolean;
+    // Parameters configured in the Modeler
+    private imageEntity: string;
+    private imageSource: ImageSource;
+    private entityConstraint: string;
+    private dataSourceMicroflow: string;
+    private captionAttr: string;
+    private descriptionAttr: string;
+    private interval: number;
+    private staticImageCollection: StaticImageCollection[];
+    private onClickEvent: OnClickEvent;
+    private callMicroflow: string;
+    private pageForm: string;
+    private pageLocation: PageLocation;
+    private width?: number;
+    private widthUnits: WidthUnits;
+    private height?: number;
+    private heightUnits: HeightUnits;
+    // Internal variables. Non-primitives created in the prototype are shared between all widget instances.
+    private contextObject: mendix.lib.MxObject;
+    private data: Data[];
+    private isLoading: boolean;
+
+    constructor(args?: Object, elem?: HTMLElement) {
+        // Do not add any default value here... it wil not run in dojo!     
+        super() ;
+        return new DojoCarousel(args, elem);
+    }
+
+    public createProps(): ImageCarouselProps {
```

Comment:
In typescript methods are `public` by default, so no need to specify that. That happens in a lot of places, so I won't repeat the comment :).